agent/commands/run.go

```diff
@@ -56,75 +55,85 @@ func Run() {
 		s := <-signals
 		signal.Stop(signals)
 		l.Warnf("Got %s, shutting down...", unix.SignalName(s.(unix.Signal))) //nolint:forcetypeassert
-		if cfg != nil {
-			cleanupTmp(cfg.Paths.TempDir, l)
-		}
-		cancel()
```

Comment:
When its cancelled it will hit cleanupTmp there: https://github.com/percona/pmm/pull/2410/files/a370bbfd23a0fa3a8a3103eaeeeaad882d32874b#diff-ae8858561a752ed8fb97d610805944a2f0123026d91e30fdbcf8c34b3ae80f5bR115